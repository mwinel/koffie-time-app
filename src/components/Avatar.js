import React from "react";

const Avatar = ({ avatar }) => {
  const styles = {
    backgroundImage: `linear-gradient(180deg, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2)), url(${avatar})`,
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    backgroundPosition: "50% 50%",
    width: "36px",
    height: "36px"
  };
  return (
    <div className="author-avatar mr-2" style={styles}>
    </div>
  );
};

export default Avatar;
