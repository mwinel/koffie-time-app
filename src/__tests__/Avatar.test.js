import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Avatar from "../components/Avatar";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders avatar component", () => {
  act(() => {
    render(<Avatar />, container);
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
