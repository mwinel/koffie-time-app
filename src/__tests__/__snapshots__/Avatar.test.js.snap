// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders avatar component 1`] = `"<div class=\\"author-avatar mr-2\\" style=\\"background-size: cover; background-repeat: no-repeat; background-position: 50% 50%; width: 36px; height: 36px;\\"></div>"`;
